File: HelpfulSense/app/src/main/java/com/andruid/magic/helpfulsense/activity/IntroActivity.java
package com.andruid.magic.helpfulsense.activity;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;
import androidx.preference.PreferenceManager;

import com.andruid.magic.helpfulsense.R;
import com.github.paolorotolo.appintro.AppIntro;
import com.github.paolorotolo.appintro.AppIntroFragment;
import com.github.paolorotolo.appintro.model.SliderPagerBuilder;

public class IntroActivity extends AppIntro {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        showSkipButton(true);
        setVibrate(true);
        setVibrateIntensity(30);
        addSlide(AppIntroFragment.newInstance(buildSliderPage("Quick actions",
                "Save custom messages to send in case of emergency with location at one tap",
                R.drawable.ic_alert, R.color.colorAccent).build()));
        addSlide(AppIntroFragment.newInstance(buildSliderPage("Battery and boot",
                "Start listening for sensors when device is booted up. Send message when " +
                        "battery low so your loved ones don't worry",
                R.drawable.ic_low_battery, R.color.colorTab1).build()));
        addSlide(AppIntroFragment.newInstance(buildSliderPage("Quick gestures",
                "Shake your phone to send emergency SMS to trusted contacts",
                R.drawable.ic_message, R.color.colorTab2).build()));
        addSlide(AppIntroFragment.newInstance(buildSliderPage("Select contacts",
                "Select contacts you trust to send your location and message to get help",
                R.drawable.ic_contacts, R.color.colorTab3).build()));
    }

    @Override
    public void onSkipPressed(Fragment currentFragment) {
        super.onSkipPressed(currentFragment);
        goToMainActivity();
    }

    @Override
    public void onDonePressed(Fragment currentFragment) {
        super.onDonePressed(currentFragment);
        goToMainActivity();
    }

    private void goToMainActivity() {
        PreferenceManager.getDefaultSharedPreferences(this)
                .edit()
                .putBoolean(getString(R.string.pref_first), false)
                .apply();
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }

    private SliderPagerBuilder buildSliderPage(String title, String desc, int image, int bgColor){
        return new SliderPagerBuilder()
                .title(title)
                .description(desc)
                .imageDrawable(image)
                .bgColor(ContextCompat.getColor(this, bgColor))
                .titleColor(ContextCompat.getColor(this, android.R.color.white))
                .descColor(ContextCompat.getColor(this, android.R.color.white));
    }
}

File: HelpfulSense/app/src/main/java/com/andruid/magic/helpfulsense/ui/activity/IntroActivity.kt
package com.andruid.magic.helpfulsense.ui.activity

import android.content.Intent
import android.os.Bundle
import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes
import androidx.fragment.app.Fragment
import com.andruid.magic.eezetensions.color
import com.andruid.magic.helpfulsense.R
import com.andruid.magic.helpfulsense.util.firstTimeDone
import com.github.appintro.AppIntro2
import com.github.appintro.AppIntroFragment
import com.github.appintro.AppIntroPageTransformerType
import com.github.appintro.model.SliderPagerBuilder

/**
 * Intro screen of the application
 */
class IntroActivity : AppIntro2() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        addSlide(AppIntroFragment.newInstance(buildSliderPage(getString(R.string.quick_actions_title),
                getString(R.string.quick_actions_desc), R.drawable.ic_alert, R.color.colorAccent).build()))
        addSlide(AppIntroFragment.newInstance(buildSliderPage(getString(R.string.autostart_title),
                getString(R.string.autostart_desc), R.drawable.ic_low_battery, R.color.colorTab1).build()))
        addSlide(AppIntroFragment.newInstance(buildSliderPage(getString(R.string.quick_gestures_title),
                getString(R.string.quick_gestures_desc), R.drawable.ic_message, R.color.colorTab2).build()))
        addSlide(AppIntroFragment.newInstance(buildSliderPage(getString(R.string.emergency_contacts_title),
                getString(R.string.emergency_contacts_desc), R.drawable.ic_contacts, R.color.colorTab3).build()))

        isSkipButtonEnabled = true
        isVibrate = true
        vibrateDuration = 30
        setTransformer(AppIntroPageTransformerType.Fade)
    }

    override fun onSkipPressed(currentFragment: Fragment?) {
        super.onSkipPressed(currentFragment)
        goToHomeActivity()
    }

    override fun onDonePressed(currentFragment: Fragment?) {
        super.onDonePressed(currentFragment)
        goToHomeActivity()
    }

    override fun onResume() {
        super.onResume()
        setImmersiveMode()
    }

    /**
     * Launch [HomeActivity] when intro is finished/skipped
     */
    private fun goToHomeActivity() {
        firstTimeDone()
        startActivity(Intent(this, HomeActivity::class.java))
        finish()
    }

    /**
     * Build a intro page instance for a feature
     * @param title title of the feature
     * @param desc description of the feature
     * @param image image resource of the feature
     * @param bgColor background color of the feature intro
     * @return intro slider for the feature
     */
    private fun buildSliderPage(title: String, desc: String, @DrawableRes image: Int, @ColorRes bgColor: Int)
            : SliderPagerBuilder {
        return SliderPagerBuilder()
                .title(title)
                .description(desc)
                .imageDrawable(image)
                .backgroundColor(color(bgColor))
                .titleColor(color(android.R.color.white))
                .descriptionColor(color(android.R.color.white))
    }
}